ca_description/urdf/dock/dock.xacro

```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" ?>
+<robot name="dock" xmlns:xacro="http://ros.org/wiki/xacro">
+  <xacro:include filename="$(find ca_description)/urdf/common_properties.xacro" />
+  <xacro:macro name="dock" params="width">
+    <link name='back'>
+      <visual name='visual'>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <box size="${width} ${width/3} ${width * 2/3}"/>
+        </geometry>
+      </visual>
+      <collision name='collision'>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <box size="${width} ${width/3} ${width * 2/3}"/>
+        </geometry>
+      </collision>
+      <xacro:inertial_cuboid_with_pose mass="0.15" x_length="${width}"
+          y_length="${width/3}" z_length="${width * 2/3}" >
+        <origin xyz="0 0 0" />
+      </xacro:inertial_cuboid_with_pose>
+    </link>
+    <link name='base'>
+      <visual name='visual'>
+        <origin xyz="0 ${width/2 - width/6} ${-(width * 2/3)/2}" rpy="0 0 0" />
+        <geometry>
+          <box size="${width} ${width} 0.01"/>
+        </geometry>
+      </visual>
+      <collision name='collision'>
+        <origin xyz="0 ${width/2 - width/6} ${-(width * 2/3)/2}" rpy="0 0 0" />
+        <geometry>
+          <box size="${width} ${width} 0.01"/>
+        </geometry>
+      </collision>
+      <xacro:inertial_cuboid_with_pose mass="0.05" x_length="${width}"
+          y_length="${width}" z_length="0.01" >
+        <origin xyz="0 ${width/2 - width/6} ${-(width * 2/3)/2}" rpy="0 0 0" />
+      </xacro:inertial_cuboid_with_pose>
+    </link>
+    <joint name='glue' type='fixed'>
+      <parent link="back" />
+      <child link="base" />
+    </joint>
+  </xacro:macro>
+  <xacro:dock width="0.15">
```

Comment:
Think about this like a library, where you create a function (in this case, the dock macro) and in the same place you use it. That's not correct. What you should do is to create a new file using this macro.